### PR TITLE
Bring in LLDP support

### PR DESCRIPTION
**Pull request by Teknoenie**
Bring in LLDP support from https://github.com/mgenov/ipxe/commit/ea0686cf7cc0dec5e5fde16ee412f7e6392b7a9e

I have tested this against the following `make` targets successfully without issue

```
make
make bin-x86_64-efi/snponly.efi
make bin-x86_64-efi/ipxe.efi
```

I've also performed a simple test and an LLDP packet was found in my environment on the interface.

Based on this I am creating this pull request.

### CODE REVIEW

File: src/include/ipxe/errfile.h.orig [hunk 1]
```diff
@@ -0,0 +1,400 @@
+#ifndef _IPXE_ERRFILE_H
```
Comment (NiKiZe) [inline src/include/ipxe/errfile.h.orig:1]: This file is probably not supposed to be here.

File: src/include/ipxe/settings.h [hunk 2]
```diff
@@ -78,6 +78,7 @@ struct setting {
 #define SETTING_AUTH_EXTRA	16 /**< Authentication additional settings */
 #define SETTING_CRYPTO		17 /**< Cryptography settings */
 #define SETTING_MISC		18 /**< Miscellaneous settings */
+#define SETTING_LLDP		15 /**< LLDP settings */
```
Comment (NiKiZe) [inline src/include/ipxe/settings.h:81]: Numbering needs fix

**Review by NiKiZe — CHANGES_REQUESTED** [review]

File: src/include/ipxe/settings.h (hunk 2 above)
Comment (Teknoenie) [reply src/include/ipxe/settings.h]: I have removed the `src/include/ipxe/errfile.h.orig` and corrected the SETTING_LINE (15->19)

**Comment by antonym** [comment]
For what it's worth, we've been using LLDP support in production for years successfully.  It would be great to have this in and have this in and not have to run a fork.  It's a great feature for large scale deployments.

**Comment by NiKiZe** [comment]
This relates to #355 
that mgenov/ipxe@ea0686c can be taken and rebased on current master.
That would conserve original commits.

**Comment by santosh0705** [comment]
Is there anyone tested this feature? I would have but don't have hardware switches to test it.
It would be an awesome feature if this got implemented in ipxe.

**Comment by santosh0705** [comment]
> We used it successfully on a very large production fleet for identification and provisioning of hosts. We had to maintain a fork to get this patch in but it works really well. I am a plus one to get it added.

Thanks @antonym for confirming.

**Comment by ITJamie** [comment]
would love to see this merged in!

**Comment by rissson** [comment]
Pinging @mareo for the initial work on this.

**Comment by Mareo** [comment]
I am glad to learn that LLDP support has been useful to some people.

I'd advise against merging it in its current state since it introduce a crash when a LLDP frame is received while an user is browsing the LLDP setting namespace in the `config` UI.

**Comment by NiKiZe** [comment]
I would like to see the original commits rebased on master rather than recreated to keep original author.

As such I do not see that this PR will be merged in its current state.
@Mareo do you think the settings logic can be reimplemented to remove the crash? Or what is the reason that it is an issue here, but not for some other settings we have that might get updated on the fly? I would also like to see your own PR on this. 😉

**Comment by Mareo** [comment]
@NiKiZe I worked a fix in my old personal repository, before it was moved to GitHub https://github.com/ipxe/people-mareo-ipxe/commit/0d5ceabc3990c0b48aa580359931cc3516ea8c29. @mcb30 had some issues with this commit and it has been stuck ever since. The UI code changed a lot since then and it is likely that this approach is not longer an option.

I would gladly rebase the "LLDP support" commit on current master and open a new PR, could you give me write access on [ipxe/people-mareo-ipxe](https://github.com/ipxe/people-mareo-ipxe)?

**Comment by mcb30** [comment]
> I would gladly rebase the "LLDP support" commit on current master and open a new PR, could you give me write access on [ipxe/people-mareo-ipxe](https://github.com/ipxe/people-mareo-ipxe)?

No need: just create the pull request from your own GitHub fork of iPXE.  The assorted `ipxe/people-*` repos are a legacy from the pre-GitHub days.

In terms of fixing the potential crash: the most robust approach is probably for the code to register a `.lldp` settings block for each network device at the time the network device is created (via a `struct net_driver`), and then to update the per-netdev saved LLDP packet whenever one is received (without unregistering and re-registering the settings block).

**Pull request closed without merging**